File: Electronics_Shop_17.Model/Requests/AddAccessory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Electronics_Shop_17.Model.Helpers;

namespace Electronics_Shop_17.Model.Requests
{
    public class AddAccessory
    {
        public string Name { get; set; }
        public string Description { get; set; }

        public int ProductId { get; set; }

        public int AccessoryCategoryId { get; set; }

        public virtual List<AddAccessoryProperty> AccessoryProperties { get; set; }

    }
}

File: angular17_web/src/app/models/accessory/add.accessory.ts
import {AddAccessoryProperty} from '../accessory-property/add.accessory.property';

export class AddAccessory {
  name!: string;
  description!: string;
  productId!: number;
  accessoryCategoryId!: number;
  accessoryProperties?: AddAccessoryProperty[];

  constructor(data?: any) {
    if (!data) return;

    this.name = data.name;
    this.description = data.description;
    this.productId = data.productId;
    this.accessoryCategoryId = data.accessoryCategoryId;
    this.accessoryProperties = Array.isArray(data.accessoryProperties)
      ? data.accessoryProperties.map((x: any) => new AddAccessoryProperty(x))
      : undefined;
  }
}
